CMakeLists.txt

```diff
@@ -306,6 +306,16 @@ set(BOUT_SOURCES
   ${CMAKE_CURRENT_BINARY_DIR}/include/bout/version.hxx
   )
 
+
+ADD_CUSTOM_COMMAND( OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/src/field/generated_fieldops.cxx
+  COMMAND python gen_fieldops.py --filename generated_fieldops.cxx.tmp
```

Comment:
If we add [`find_package(Python)`](https://cmake.org/cmake/help/latest/module/FindPython.html) we can using `Python_EXECUTABLE` to make this nice and portable. I think that also prefers to find python 3 first too, which is nice.

The same package can also find the numpy include headers, which might be useful for building boutcore?